c_binding/include/libstoragemgmt/libstoragemgmt.h

```diff
@@ -224,63 +439,164 @@ int LSM_DLL_EXPORT lsm_job_status_ss_get(lsm_connect *c,
                                          lsm_fs_ss ** ss, lsm_flag flags);
 
 /**
- * Frees the resources used by a job.
- * @param[in] conn          Valid connection pointer
- * @param[in] job_id        Job ID
- * @param[in] flags         Reserved for future use, must be zero.
- * @return Error code as enumerated by \ref lsm_error_number.
- * @retval LSM_ERR_OK on success.
+ * lsm_job_free - Frees the resources used by a job.
+ *
+ * Version:
+ *      1.0
+ *
+ * Description:
+ *      Frees the resources used by a job.
+ *
+ * @conn:
+ *      Valid connection pointer
+ * @job_id:
+ *      String. Job ID
+ * @flags:
+ *      Reserved for future use, must be LSM_CLIENT_FLAG_RSVD.
+ *
+ * Return:
+ *      Error code as enumerated by 'lsm_error_number'.
+ *          * LSM_ERR_OK
+ *              On success.
+ *          * LSM_ERR_INVALID_ARGUMENT
+ *              When any argument is NULL or not a valid lsm_connect pointer
+ *              or invalid flags.
+ *          * LSM_ERR_NOT_FOUND_JOB
+ *              When job not found.
+ *
  */
 int LSM_DLL_EXPORT lsm_job_free(lsm_connect *conn, char **job_id,
                                 lsm_flag flags);
-/**
- * Storage system query functions
- */
 
 /**
- * Query the capabilities of the storage array.
- * @param[in]   conn    Valid connection @see lsm_connect_password
- * @param[in]   system  System of interest
- * @param[out]  cap     The storage array capabilities
- * @param[in] flags     Reserved for future use, must be zero.
- * @return Error code as enumerated by \ref lsm_error_number.
- * @retval LSM_ERR_OK on success.
+ * lsm_capabilities - Query the capabilities of the storage array.
+ *
+ * Version:
+ *      1.0
+ *
+ * Description:
+ *      Query the capabilities of the storage array.
+ *      Capability is used to indicate whether certain function is supported
+ *      by specified storage system. Please check desired function for required
+ *      capability. To verify certain capability is supported, use
+ *      lsm_capability_supported(). If the document any function does specified
+ *      required capability, that means it's mandatory supported.
+ *
```

Comment:
If the functionality is not listed in the enumerated capability type (lsm_capabilty_type) then that functionality is mandatory and required to exist.